### PR TITLE
chore(wallet): Upgrade react-native to 0.71.7

### PR DESCRIPTION
**Pull request by coreyphillips**
# Description

- Upgrades `react-native` to `0.71.7`.
- Upgrades dependencies in `package.json`.
- Android is now using `hermes` instead of the `v8` engine.

## Type of change
- [x] Dependency Upgrade (non-breaking changes which adds functionality)

## QA Notes

- Prior to running Bitkit we need to first specify either `yarn setup-ios` or `yarn setup-android`.
    - This will be removed once we have a stable `libsodium` branch for both iOS & Android.
- Please suggest changes or create PR's against this branch if you see anything that needs fixing or additional attention.
- All functionality throughout the app should be tested to ensure that functionality is retained or improved.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +23,7 @@
     "lint:check": "eslint . --ext .js,.jsx,.ts,.tsx",
     "lint:fix": "eslint . --fix --ext .js,.jsx,.ts,.tsx",
     "tsc:check": "tsc -p tsconfig.json --noEmit",
-    "clean": "yarn clean:rn && yarn clean:ios && yarn clean:android && yarn start --reset-cache",
+    "clean": "yarn clean:rn && yarn clean:ios && yarn start --reset-cache",
```
Comment (pwltr) [inline package.json]: Revert this?

File: BUILD.md [hunk 2]
```diff
@@ -18,6 +18,30 @@ Switch to the Node.js version defined in `.nvmrc`. If `nvm` (or similiar) is ins
...
+
+```shell
+yarn ios
+```
+
+or
+
+```shell
+yarn android
+```
```
Comment (pwltr) [inline BUILD.md]: No "Run" step needed in `Build.md`

File: BUILD.md [hunk 3]
```diff
@@ -18,6 +18,30 @@ Switch to the Node.js version defined in `.nvmrc`. If `nvm` (or similiar) is ins
...
+
+```shell
+yarn setup-ios
+```
+
+or
+
+```shell
+yarn setup-android
+```
```
Comment (pwltr) [inline BUILD.md:31]: Also add this to `Development.md`

File: package.json [hunk 4]
```diff
@@ -65,137 +67,151 @@
...
+    "immer": "^9.0.19",
+    "jdenticon": "^3.2.0",
     "lodash.clonedeep": "^4.5.0",
     "lodash.debounce": "^4.0.8",
     "lodash.isequal": "^4.5.0",
     "lodash.throttle": "^4.1.1",
-    "lottie-react-native": "^5.1.3",
-    "nodejs-mobile-react-native": "^0.7.0",
+    "lottie-react-native": "^5.1.5",
+    "nodejs-mobile-react-native": "https://github.com/Shopify/nodejs-mobile-react-native",
```
Comment (limpbrains) [inline package.json]: I think we should pin commit hash here, so we don't upgrade it accidently

File: package.json [hunk 5]
```diff
@@ -65,137 +67,151 @@
...
     "rn-nodeify": "^10.3.0",
     "semver": "^7.3.8",
-    "typescript": "^4.7.4"
+    "typescript": "4.8.4"
+  },
+  "reactNativePermissionsIOS": [
+    "Camera",
+    "Contacts",
+    "FaceID",
+    "LocationWhenInUse",
```
Comment (limpbrains) [inline package.json]: do we need location?

File: ios/bitkit/AppDelegate.mm [hunk 6]
```diff
@@ -1,70 +1,18 @@
...
-  self.window.rootViewController = rootViewController;
-  [self.window makeKeyAndVisible];
-
-  // adding it for boot splash BEGIN
-  UIStoryboard *launchScreenStoryboard = [UIStoryboard storyboardWithName:@"LaunchScreen" bundle:nil];
-  UIViewController *launchScreenViewController = [launchScreenStoryboard instantiateViewControllerWithIdentifier:@"LaunchScreenViewController"];
-  UIView *launchScreenView = [launchScreenViewController view];
-  launchScreenView.frame = self.window.bounds;
-  rootView.loadingView = launchScreenView;
-  // adding it for boot splash END
```
Comment (limpbrains) [inline ios/bitkit/AppDelegate.mm]: We need to try to keep it, it shows logo while running RN. I will try to do it

File: _node-version
Comment (pwltr) [inline _node-version:1]: What is this file for? I usually do `nvm use` and it sets the right version for the project. You using something else?

**Comment by pwltr** [comment]
Very nice to see this. Has anyone managed to build on Android? I'm getting
```
error: cannot find symbol
import com.bitkit.R;
```
both on Linux and MacOS.

**Review by pwltr — CHANGES_REQUESTED** [review]

File: _node-version
Comment (coreyphillips) [reply _node-version:1]: This file was created and included during the `npx react-native init bitkit` command.

File: package.json (hunk 1 above)
Comment (coreyphillips) [reply package.json]: Reverted [here](https://github.com/synonymdev/bitkit/pull/982/commits/0ef1f88599bdc827efae4a7a4490924337d23777).

File: BUILD.md (hunk 2 above)
Comment (coreyphillips) [reply BUILD.md]: Updated [here](https://github.com/synonymdev/bitkit/pull/982/commits/0ef1f88599bdc827efae4a7a4490924337d23777).

File: BUILD.md (hunk 3 above)
Comment (coreyphillips) [reply BUILD.md:31]: Updated [here](https://github.com/synonymdev/bitkit/pull/982/commits/0ef1f88599bdc827efae4a7a4490924337d23777).

**Comment by coreyphillips** [comment]
> Very nice to see this. Has anyone managed to build on Android? I'm getting
> 
> ```
> error: cannot find symbol
> import com.bitkit.R;
> ```
> 
> both on Linux and MacOS.

Prior to running Android are you running `yarn setup-android`? Perhaps `yarn setup-android && yarn clean && yarn android` if the `setup-android` command alone is not working.

**Comment by pwltr** [comment]
> Prior to running Android are you running `yarn setup-android`? Perhaps `yarn setup-android && yarn clean && yarn android` if the `setup-android` command alone is not working.

No luck, here's the full output of `yarn android`:

```
yarn run v1.22.19
$ react-native run-android
warn Package nodejs-mobile-react-native contains invalid configuration: "dependency.platforms.ios.project" is not allowed. Please verify it's properly linked using "react-native config" command and contact the package maintainers about this.
info JS server already running.
info Installing the app...

> Configure project :lottie-react-native
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

> Configure project :react-native-flipper
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

> Configure project :react-native-rate
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

> Configure project :react-native-reanimated
AAR for react-native-reanimated has been found
/Users/phil/Dev/synonymdev/bitkit-test/node_modules/react-native-reanimated/android/react-native-reanimated-71-hermes.aar

> Configure project :shopify_react-native-skia
react-native-skia: node_modules/ found at: /Users/phil/Dev/synonymdev/bitkit-test/node_modules
react-native-skia: RN Version: 71 / 0.71.6
react-native-skia: isSourceBuild: false
react-native-skia: PrebuiltDir: /Users/phil/Dev/synonymdev/bitkit-test/node_modules/@shopify/react-native-skia/android/build/react-native-0*/jni
react-native-skia: buildType: debug
react-native-skia: buildDir: /Users/phil/Dev/synonymdev/bitkit-test/node_modules/@shopify/react-native-skia/android/build
react-native-skia: node_modules: /Users/phil/Dev/synonymdev/bitkit-test/node_modules
react-native-skia: Enable Prefab: true
react-native-skia: aar state post 70, do nothing
WARNING:The specified Android SDK Build Tools version (28.0.3) is ignored, as it is below the minimum supported version (30.0.3) for Android Gradle Plugin 7.4.2.
Android SDK Build Tools 30.0.3 will be used.
To suppress this warning, remove "buildToolsVersion '28.0.3'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.

> Task :app:compileDebugJavaWithJavac FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings
436 actionable tasks: 3 executed, 433 up-to-date
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/modules/SplashScreen/SplashScreenModule.java:9: error: cannot find symbol
import com.bitkit.R;
                 ^
  symbol:   class R
  location: package com.bitkit
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/MainApplication.java:20: error: cannot find symbol
          return BuildConfig.DEBUG;
                 ^
  symbol: variable BuildConfig
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/MainApplication.java:40: error: cannot find symbol
          return BuildConfig.IS_NEW_ARCHITECTURE_ENABLED;
                 ^
  symbol: variable BuildConfig
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/MainApplication.java:45: error: cannot find symbol
          return BuildConfig.IS_HERMES_ENABLED;
                 ^
  symbol: variable BuildConfig
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/MainApplication.java:58: error: cannot find symbol
    if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED) {
        ^
  symbol:   variable BuildConfig
  location: class MainApplication
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/MainApplication.java:58: error: illegal parenthesized expression
    if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED) {
       ^
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/MainActivity.java:24: error: package R does not exist
    setTheme(R.style.AppTheme);
              ^
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/modules/SplashScreen/SplashScreenModule.java:72: error: package R does not exist
            splashDialog = new Dialog(activity, R.style.AppTheme_SplashDialog);
                                                 ^
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/modules/SplashScreen/SplashScreenModule.java:73: error: package R does not exist
            splashDialog.setContentView(R.layout.splash_screen);
                                         ^
9 errors

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
java.lang.StackOverflowError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 11s

error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup.
Error: Command failed: ./gradlew app:installDebug -PreactNativeDevServerPort=8081
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/modules/SplashScreen/SplashScreenModule.java:9: error: cannot find symbol
import com.bitkit.R;
                 ^
  symbol:   class R
  location: package com.bitkit
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/MainApplication.java:20: error: cannot find symbol
          return BuildConfig.DEBUG;
                 ^
  symbol: variable BuildConfig
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/MainApplication.java:40: error: cannot find symbol
          return BuildConfig.IS_NEW_ARCHITECTURE_ENABLED;
                 ^
  symbol: variable BuildConfig
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/MainApplication.java:45: error: cannot find symbol
          return BuildConfig.IS_HERMES_ENABLED;
                 ^
  symbol: variable BuildConfig
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/MainApplication.java:58: error: cannot find symbol
    if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED) {
        ^
  symbol:   variable BuildConfig
  location: class MainApplication
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/MainApplication.java:58: error: illegal parenthesized expression
    if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED) {
       ^
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/MainActivity.java:24: error: package R does not exist
    setTheme(R.style.AppTheme);
              ^
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/modules/SplashScreen/SplashScreenModule.java:72: error: package R does not exist
            splashDialog = new Dialog(activity, R.style.AppTheme_SplashDialog);
                                                 ^
/Users/phil/Dev/synonymdev/bitkit-test/android/app/src/main/java/com/bitkit/modules/SplashScreen/SplashScreenModule.java:73: error: package R does not exist
            splashDialog.setContentView(R.layout.splash_screen);
                                         ^
9 errors

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
java.lang.StackOverflowError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 11s

    at makeError (/Users/phil/Dev/synonymdev/bitkit-test/node_modules/execa/index.js:174:9)
    at /Users/phil/Dev/synonymdev/bitkit-test/node_modules/execa/index.js:278:16
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async runOnAllDevices (/Users/phil/Dev/synonymdev/bitkit-test/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/runOnAllDevices.js:82:7)
    at async Command.handleAction (/Users/phil/Dev/synonymdev/bitkit-test/node_modules/@react-native-community/cli/build/index.js:108:9)
info Run CLI with --verbose flag for more details.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

and my output of `./android/gradlew --version`

------------------------------------------------------------
Gradle 7.5.1
------------------------------------------------------------

Build time:   2022-08-05 21:17:56 UTC
Revision:     d1daa0cbf1a0103000b71484e1dbfe096e095918

Kotlin:       1.6.21
Groovy:       3.0.10
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          11.0.18 (Azul Systems, Inc. 11.0.18+10-LTS)
OS:           Mac OS X 13.3.1 aarch64

**Comment by coreyphillips** [comment]
@pwltr Found the issue. Sorry about that. Should be sorted as of [this](https://github.com/synonymdev/bitkit/pull/982/commits/a29d37fe1e1a60ca9ce3343ac72f81320a8f272a) commit.

**Comment by pwltr** [comment]
> @pwltr Found the issue. Sorry about that. Should be sorted as of [this](https://github.com/synonymdev/bitkit/pull/982/commits/a29d37fe1e1a60ca9ce3343ac72f81320a8f272a) commit.

Good now! I'm seeing a performance improvement and smoother animations on Android. 🤩 
No issues so far in terms of breakage in the app. Just need to get the tests to run.

**Comment by pwltr** [comment]
App is crashing on the WalletDetails screen (iOS & Android)

image

**Comment by pwltr** [comment]
E2E tests are working in debug mode on iOS. Not for release though. Looks like something to do with Flipper.

**Comment by limpbrains** [comment]
I can't compile it in Release configuration for iOS:

```
> npx react-native run-ios --simulator="iPhone 14" --configuration Release

▸ Compiling bitkit_vers.c
▸ Compiling bitkit_vers.c
▸ Linking bitkit
⚠️  ld: Could not find or use auto-linked library 'libldk.a'

❌  ld: symbol(s) not found for architecture x86_64

❌  clang: error: linker command failed with exit code 1 (use -v to see invocation)

▸ Linking bitkit
⚠️  ld: Could not find or use auto-linked library 'libldk.a'

❌  ld: symbol(s) not found for architecture arm64
```

File: ios/bitkit/AppDelegate.mm (hunk 6 above)
Comment (limpbrains) [reply ios/bitkit/AppDelegate.mm]: fixed

**Comment by coreyphillips** [comment]
@Jasonvdb, any thoughts on why we could be getting [this](https://github.com/synonymdev/bitkit/pull/982#issuecomment-1520468696) error?

**Comment by Jasonvdb** [comment]
> @Jasonvdb, any thoughts on why we could be getting [this](https://github.com/synonymdev/bitkit/pull/982#issuecomment-1520468696) error?

Looking into it today. It comes up a lot but can be related to many different issues.

**Review by limpbrains — APPROVED** [review]

**Comment by limpbrains** [comment]
> I can't compile it in Release configuration for iOS:
> 
> ```
> > npx react-native run-ios --simulator="iPhone 14" --configuration Release
> 
> ▸ Compiling bitkit_vers.c
> ▸ Compiling bitkit_vers.c
> ▸ Linking bitkit
> ⚠️  ld: Could not find or use auto-linked library 'libldk.a'
> 
> ❌  ld: symbol(s) not found for architecture x86_64
> 
> ❌  clang: error: linker command failed with exit code 1 (use -v to see invocation)
> 
> ▸ Linking bitkit
> ⚠️  ld: Could not find or use auto-linked library 'libldk.a'
> 
> ❌  ld: symbol(s) not found for architecture arm64
> ```

I was able to compile in release mode with `NO_FLIPPER=1`
```sh
rm -rf node_modules
NO_FLIPPER=1 yarn
npx react-native run-ios --simulator="iPhone 14" --configuration Release
```

**Review by pwltr — APPROVED** [review]

**Review by Jasonvdb — APPROVED** [review]

**Pull request merged**